mim/commands/install.py

```diff
@@ -1,580 +1,303 @@
-            target_pkg = 'mmcv-full'
-
-        echo_success(f'installing {target_pkg} from local repo.')
-
-        install_from_repo(
-            repo_root,
-            package=target_pkg,
-            timeout=timeout,
-            is_yes=is_yes,
-            is_user_dir=is_user_dir,
-            is_editable=is_editable)
-
-    elif find_url and find_url.find('git') >= 0 or is_install_master:
-        install_from_github(target_pkg, target_version, find_url, timeout,
-                            is_yes, is_user_dir, is_install_master)
-    else:
-        # if installing from wheel failed, it will try to install package by
-        # building from source if possible.
-        try:
-            install_from_wheel(target_pkg, target_version, find_url, timeout,
-                               is_user_dir)
-        except RuntimeError as error:
-            if target_pkg in PKG2PROJECT:
-                find_url = f'{DEFAULT_URL}/{PKG2PROJECT[target_pkg]}.git'
-                if target_version:
-                    target_pkg = f'{target_pkg}=={target_version}'
-                if is_yes:
-                    install(target_pkg, find_url, timeout, is_yes, is_user_dir)
-                else:
-                    confirm_msg = (f'install {target_pkg} from wheel, but it '
-                                   'failed. Do you want to build it from '
-                                   'source if possible?')
-                    if click.confirm(confirm_msg):
-                        install(target_pkg, find_url, timeout, is_yes,
-                                is_user_dir)
-                    else:
-                        raise RuntimeError(
-                            highlighted_error(
-                                f'Failed to install {target_pkg}.'))
-            else:
-                raise RuntimeError(highlighted_error(error))
-
-    echo_success(f'Successfully installed {target_pkg}.')
-
-
-def looks_like_path(name: str) -> bool:
-    """Checks whether the string "looks like" a path on the filesystem.
-
-    This does not check whether the target actually exists, only judge from the
-    appearance.
+def install(
+    install_args: List[str],
+    index_url: Optional[str] = None,
+    is_yes: bool = False,
+) -> Any:
+    """Install the package via pip and add 'mim' extra requirements for
+    OpenMMLab package during pip install process.
 
     Args:
-        name (str): The string to be checked.
+        install_args (list): List of arguments passed to `pip install`.
+        index_url (str, optional): The pypi index url.
+        is_yes (bool, optional): Deprecated, will have no effect. Reserved for
+            interface compatibility only.
     """
-    if osp.sep in name:
-        return True
-    if osp.altsep is not None and osp.altsep in name:
-        return True
-    if name.startswith('.'):
-        return True
-    return False
+    # add mmcv-full find links by default
+    install_args += ['-f', get_mmcv_full_find_link()]
```

Comment:
What will it happen when install_args contains `-f` option?

Reply:
--find-links is a multiple option, see [here](https://github.com/pypa/pip/blob/6c2013cfaa42d5a53c943e72f9be627ed71569e1/src/pip/_internal/cli/cmdoptions.py#L379)

we can add multiple find links like:
`pip install xx -f <findlink1> -f <findlink2>`

Reply:
OK